hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/bucket/BucketCache.java

```diff
@@ -1053,6 +1065,12 @@ private void persistToFile() throws IOException {
     }
     try (FileOutputStream fos = new FileOutputStream(persistencePath, false)) {
       fos.write(ProtobufMagic.PB_MAGIC);
+      byte[] checksum = ((PersistentIOEngine) ioEngine).calculateChecksum(algorithm);
```

Comment:
This is not correct.
In master we already have the PB_MAGIC and we are writing the meta info into the persisting file in protobuf way.  We can change the PB proto message to include optional checksum also. If the checksum is there, write those bytes. Marking it as optional so retrieving it back as bytes and check whether it was there or not, all these work PB will help us. We dont have to have any extra logic.